contracts/ExampleToken.cdc

```diff
@@ -44,228 +81,143 @@ pub contract ExampleToken: FungibleToken {
-        ///
-        pub fun burnTokens(from: @FungibleToken.Vault) {
-            let vault <- from as! @ExampleToken.Vault
-            let amount = vault.balance
-            destroy vault
-            emit TokensBurned(amount: amount)
-        }
+    /// Function that creates a new Vault with a balance of zero
+    /// and returns it to the calling context. A user must call this function
+    /// and store the returned Vault in their storage in order to allow their
+    /// account to be able to receive deposits of this token type.
+    ///
+    access(all) fun createEmptyVault(vaultType: Type): @ExampleToken.Vault {
+        return <- create Vault(balance: 0.0)
     }
 
     init() {
         self.totalSupply = 1000.0
+
         self.VaultStoragePath = /storage/exampleTokenVault
-        self.VaultPublicPath = /public/exampleTokenMetadata
+        self.VaultPublicPath = /public/exampleTokenVault
         self.ReceiverPublicPath = /public/exampleTokenReceiver
-        self.AdminStoragePath = /storage/exampleTokenAdmin
+        self.AdminStoragePath = /storage/exampleTokenAdmin 
 
-        // Create the Vault with the total supply of tokens and save it in storage.
+        // Create the Vault with the total supply of tokens and save it in storage
+        //
         let vault <- create Vault(balance: self.totalSupply)
-        self.account.save(<-vault, to: self.VaultStoragePath)
 
         // Create a public capability to the stored Vault that exposes
-        // the `deposit` method through the `Receiver` interface.
-        self.account.link<&{FungibleToken.Receiver}>(
-            self.ReceiverPublicPath,
-            target: self.VaultStoragePath
-        )
-
-        // Create a public capability to the stored Vault that only exposes
-        // the `balance` field and the `resolveView` method through the `Balance` interface
-        self.account.link<&ExampleToken.Vault{FungibleToken.Balance}>(
-            self.VaultPublicPath,
-            target: self.VaultStoragePath
-        )
-
-        let admin <- create Administrator()
-        self.account.save(<-admin, to: self.AdminStoragePath)
-
-        // Emit an event that shows that the contract was initialized
-        emit TokensInitialized(initialSupply: self.totalSupply)
+        // the `deposit` method and getAcceptedTypes method through the `Receiver` interface
+        // and the `balance` method through the `Balance` interface
+        //
+        let exampleTokenCap = self.account.capabilities.storage.issue<&ExampleToken.Vault>(self.VaultStoragePath)
+        self.account.capabilities.publish(exampleTokenCap, at: self.VaultPublicPath)
+        let receiverCap = self.account.capabilities.storage.issue<&ExampleToken.Vault>(self.VaultStoragePath)
+        self.account.capabilities.publish(receiverCap, at: self.ReceiverPublicPath)
+
+        self.account.storage.save(<-vault, to: /storage/exampleTokenVault)
```

Comment:
Might want to emit `TokensMinted` if the initial balance is above zero.

Reply:
Updated